File: 07/claude-opus-4.5/go/main.go
package main

import (
	"fmt"
	"os"
	"strings"
)

func main() {
	data, _ := os.ReadFile("input.txt")
	input := strings.TrimSpace(string(data))
	lines := strings.Split(input, "\n")

	rows := len(lines)
	cols := 0
	if rows > 0 {
		cols = len(lines[0])
	}

	// Find starting position S
	startRow, startCol := 0, 0
	for r := 0; r < rows; r++ {
		c := strings.Index(lines[r], "S")
		if c != -1 {
			startRow, startCol = r, c
			break
		}
	}

	// Part 1: Count total splits
	var splitCount int64 = 0
	currentBeams := make(map[int]bool)
	currentBeams[startCol] = true

	for row := startRow + 1; row < rows && len(currentBeams) > 0; row++ {
		nextBeams := make(map[int]bool)
		lineLen := len(lines[row])
		for col := range currentBeams {
			if col < 0 || col >= lineLen {
				continue
			}
			cell := lines[row][col]
			if cell == '^' {
				splitCount++
				if col-1 >= 0 {
					nextBeams[col-1] = true
				}
				if col+1 < cols {
					nextBeams[col+1] = true
				}
			} else {
				nextBeams[col] = true
			}
		}
		currentBeams = nextBeams
	}

	fmt.Printf("Day 07 Part 1: %d\n", splitCount)

	// Part 2: Count timelines (each particle takes both paths at each splitter)
	particleCounts := make(map[int]int64)
	particleCounts[startCol] = 1

	for row := startRow + 1; row < rows && len(particleCounts) > 0; row++ {
		nextCounts := make(map[int]int64)
		lineLen := len(lines[row])
		for col, count := range particleCounts {
			if col < 0 || col >= lineLen {
				continue
			}
			cell := lines[row][col]
			if cell == '^' {
				if col-1 >= 0 {
					nextCounts[col-1] += count
				}
				if col+1 < cols {
					nextCounts[col+1] += count
				}
			} else {
				nextCounts[col] += count
			}
		}
		particleCounts = nextCounts
	}

	var part2 int64 = 0
	for _, count := range particleCounts {
		part2 += count
	}
	fmt.Printf("Day 07 Part 2: %d\n", part2)
}

File: 07/claude-opus-4.5/rust/src/main.rs
use std::collections::HashSet;
use std::fs;

fn main() {
    let input = fs::read_to_string("input.txt")
        .expect("Failed to read input");
    let lines: Vec<&str> = input.trim().lines().collect();

    let rows = lines.len();
    let cols = if rows > 0 { lines[0].len() } else { 0 };

    // Find starting position S
    let mut start_row = 0;
    let mut start_col = 0;
    for (r, line) in lines.iter().enumerate() {
        if let Some(c) = line.find('S') {
            start_row = r;
            start_col = c;
            break;
        }
    }

    // Part 1: Count total splits
    let mut split_count: i64 = 0;
    let mut current_beams: HashSet<usize> = HashSet::new();
    current_beams.insert(start_col);

    for row in (start_row + 1)..rows {
        if current_beams.is_empty() {
            break;
        }
        let mut next_beams: HashSet<usize> = HashSet::new();
        for &col in &current_beams {
            if col >= cols {
                continue;
            }
            let cell = lines[row].as_bytes()[col];
            if cell == b'^' {
                split_count += 1;
                if col > 0 {
                    next_beams.insert(col - 1);
                }
                if col + 1 < cols {
                    next_beams.insert(col + 1);
                }
            } else {
                next_beams.insert(col);
            }
        }
        current_beams = next_beams;
    }

    println!("Day 07 Part 1: {}", split_count);

    // Part 2: Count timelines (each particle takes both paths at each splitter)
    let mut particle_counts: std::collections::HashMap<usize, i64> = std::collections::HashMap::new();
    particle_counts.insert(start_col, 1);

    for row in (start_row + 1)..rows {
        if particle_counts.is_empty() {
            break;
        }
        let mut next_counts: std::collections::HashMap<usize, i64> = std::collections::HashMap::new();
        for (&col, &count) in &particle_counts {
            if col >= cols {
                continue;
            }
            let cell = lines[row].as_bytes()[col];
            if cell == b'^' {
                if col > 0 {
                    *next_counts.entry(col - 1).or_insert(0) += count;
                }
                if col + 1 < cols {
                    *next_counts.entry(col + 1).or_insert(0) += count;
                }
            } else {
                *next_counts.entry(col).or_insert(0) += count;
            }
        }
        particle_counts = next_counts;
    }

    let part2: i64 = particle_counts.values().sum();
    println!("Day 07 Part 2: {}", part2);
}
